server/controllers/lists.js

```diff
@@ -6,6 +6,29 @@ class ListController {
         this.database = database;
     }
 
+    async getListsNearCoordinates(req, res, next) {
+        if (!req.body.latitude || !req.body.longitude){
+            return;
+        }
+
+        const selectQuery = `SELECT * FROM lists
+        WHERE status="${ListController.LIST_STATUS.UNCLAIMED}"
+        AND latitude BETWEEN ${req.body.latitude - 0.5} AND ${req.body.latitude + 0.5}
+        AND longitude BETWEEN ${req.body.longitude - 0.5} AND ${req.body.longitude + 0.5}
```

Comment:
I could inject all the SQL up in here. Also if the lat/long come in as strings they'd get concatted a + 0.5. Both of those problems could be fixed by casting the parameters as a number.

```suggestion
        AND latitude BETWEEN ${Number(req.body.latitude) - 0.5} AND ${Number(req.body.latitude) + 0.5}
        AND longitude BETWEEN ${Number(req.body.longitude) - 0.5} AND ${Number(req.body.longitude) + 0.5}
```

Reply:
that seems like a thing i should do then!